### PR TITLE
[Refactor]Cleanup the thrust invocations.

### PR DESCRIPTION
**Pull request by bashimao**
This is the isolated change where I cleanup all the thrust invocations.

The type you were looking for is named: uintptr_t

https://en.cppreference.com/w/cpp/types/integer

### CODE REVIEW

File: include/merlin_hashtable.cuh [hunk 1]
```diff
@@ -308,22 +314,13 @@ class HashTable {
...
 
       {
-        static_assert(sizeof(value_type*) == sizeof(uint64_t),
-                      "[HierarchicalKV] illegal conversation. value_type pointer "
-                      "should be 64 bit!");
-
         const size_t N = n;
-        thrust::device_ptr<uint64_t> d_dst_ptr(
-            reinterpret_cast<uint64_t*>(d_dst));
+        thrust::device_ptr<uintptr_t> d_dst_ptr(
```
Comment (rhdong) [inline include/merlin_hashtable.cuh:318]: Pls add the header of <cstdint>for the uintptr_t.

**Comment by rhdong** [comment]
Hi @bashimao, please squash small commits into one, thank you!

File: include/merlin_hashtable.cuh (hunk 1 above)
Comment (bashimao) [reply include/merlin_hashtable.cuh:319]: Done!

**Comment by bashimao** [comment]
Sqashed!

**Review by rhdong — APPROVED** [review]
LGTM

**Pull request merged**